# HW 1 - Sentiment Analysis on Twitter

### Objective: To use iterative python skills to perform a sentiment analysis on a large database of tweets

#### Section 1: Counting Words in Each Tweet

##### Section 1.1: Trump-related Tweets

Our first step will be to read the txt files into a python string. We have a three, each containing a list of positive, negative and stop words. We also have one more titled Trump.txt containing tweets containing the keyword 'Trump' which we extracted in real-time using a twitter development account. 

In [1]:
pos = open('positive.txt','r')
poscontent = pos.read()  #text of all positive words (not in list form yet due to messiness)
pos.close() 

neg = open('negative.txt','r')
negcontent = neg.read()  #text of all negative words (not in list form yet due to messiness)
neg.close() 

stop = open('stopwords.txt','r')
stopcontent = stop.read()  #text of all stop words (not in list form yet due to messiness)
stop.close() 

trump = open('Trump.txt','r')
trump_content = trump.read()  #text of all trump-related tweets (not in list form yet due to messiness)
trump.close() 

Let's split the positive, negative, and stopwords into a list. It's easy to do since they're delimited only by a newline character. 

In [2]:
poslist=poscontent.split('\n')
neglist=negcontent.split('\n')
stoplist=stopcontent.split('\n')  #Here we're making a list delimited at the newlines characters/ removing newline

In [3]:
trump_content   # raw text of all trump related tweets (not yet in list form)

'b\'RT @DanRather: Donald Trump likes to boast about being a businessman. But how long do you think someone acting and talking like him would l\\xe2\\x80\\xa6\\n\'b\'RT @maggieNYT: \\xe2\\x80\\x9cAnother former senior official said it was a constant struggle to convince Mr. Trump that Russia, not Ukraine, had interfer\\xe2\\x80\\xa6\\n\'b\'RT @ChrisMurphyCT: The transcript of the Zelensky call is an admission of guilt.  \\n\\nThe whistle blower complaint is credible and adds some\\xe2\\x80\\xa6\\n\'b\'RT @LiteraryMouse: @RepKinzinger You need to call for Trump\\xe2\\x80\\x99s resignation. He\\xe2\\x80\\x99s making veiled death threats against the whistleblower and o\\xe2\\x80\\xa6\\n\'b\'RT @BryanDawsonUSA: Jill Lepore in the @NewYorker in 2018\\n\\n\\xe2\\x80\\x9cWhat else has [Trump] done? Trump\\xe2\\x80\\x99s entire presidency, from his cabinet appoint\\xe2\\x80\\xa6\\n\'b\'RT @jasoninthehouse: ,@RepAdamSchiff should be censured by the House https://t.co/eOTkU41IcL\\n\'b\'RT @Kaspa

Now let's use some string methods to clean up the messy list of tweets displayed above. We will clean as much as possible. 

It's important to make the entire string lowercase, as well as to remove any unnecessary puncuation. This is important, because when counting the instances of positive, negative, and stopwords; any puncuation or case difference could confuse our counter. 

<b>For example: *Good* $\neq$ *good* $\neq$ *good!*</b>

In [4]:
#Let's use some string methods to turn all words 
#into lower case and remove puncuation so as to remove all unnessecary noise

trump_content=trump_content.lower()


badchars='.,!?"///#[]()*$:'
for i in badchars:
    trump_content=trump_content.replace(i,'')

The string of tweets should now be turned into a list of words. This way we can iterate through all the words and count how many positives and negatives appear. 

In [5]:
trumplist=trump_content.split()
trumplist  # here we turned the trump text into a list of words. 

["b'rt",
 '@danrather',
 'donald',
 'trump',
 'likes',
 'to',
 'boast',
 'about',
 'being',
 'a',
 'businessman',
 'but',
 'how',
 'long',
 'do',
 'you',
 'think',
 'someone',
 'acting',
 'and',
 'talking',
 'like',
 'him',
 'would',
 "l\\xe2\\x80\\xa6\\n'b'rt",
 '@maggienyt',
 '\\xe2\\x80\\x9canother',
 'former',
 'senior',
 'official',
 'said',
 'it',
 'was',
 'a',
 'constant',
 'struggle',
 'to',
 'convince',
 'mr',
 'trump',
 'that',
 'russia',
 'not',
 'ukraine',
 'had',
 "interfer\\xe2\\x80\\xa6\\n'b'rt",
 '@chrismurphyct',
 'the',
 'transcript',
 'of',
 'the',
 'zelensky',
 'call',
 'is',
 'an',
 'admission',
 'of',
 'guilt',
 '\\n\\nthe',
 'whistle',
 'blower',
 'complaint',
 'is',
 'credible',
 'and',
 'adds',
 "some\\xe2\\x80\\xa6\\n'b'rt",
 '@literarymouse',
 '@repkinzinger',
 'you',
 'need',
 'to',
 'call',
 'for',
 'trump\\xe2\\x80\\x99s',
 'resignation',
 'he\\xe2\\x80\\x99s',
 'making',
 'veiled',
 'death',
 'threats',
 'against',
 'the',
 'whistleblower',
 'and',
 "o\\x

In the preview above, we still see lots of unnecessary noise. Let's use some further string methods to remove any web links, usernames, retweets, and what not. 

In [6]:
import random
random.seed(1)   #set seed for consistent number generation
for i in trumplist:
    if i.startswith("https") or i.endswith("b'rt") or i.startswith("@") or "\\" in i or 'xe2' in i:
        trumplist.remove(i)
        
#Use the startswith() and endswith() methods to remove noise. Since user names begin with '@' 
# and links begin with 'http', etc


for i in trumplist:
    if i.startswith("https"):
        trumplist.remove(i)
        
    if i.startswith("@"):
        trumplist.remove(i)
        

#Loop one more time to make sure everything is accounted for 

In [7]:

for i in trumplist:
    print(i)  #Most of the noise has been removed

donald
trump
likes
to
boast
about
being
a
businessman
but
how
long
do
you
think
someone
acting
and
talking
like
him
would
former
senior
official
said
it
was
a
constant
struggle
to
convince
mr
trump
that
russia
not
ukraine
had
the
transcript
of
the
zelensky
call
is
an
admission
of
guilt
whistle
blower
complaint
is
credible
and
adds
you
need
to
call
for
resignation
making
veiled
death
threats
against
the
whistleblower
and
jill
lepore
in
the
in
else
has
trump
done
entire
presidency
from
his
cabinet
should
be
censured
by
the
house
as
you
watch
defenders
lie
deflect
and
distract
today
and
in
the
coming
weeks
remember
that
they
care
by
calling
for
a
civil
war
tonight
donald
trump
has
committed
treason
he
may
have
also
violated
terms
of
man
was
investigating
epstien
clinton
biden
trump
an
american
citizen
is
in
danger
he
needs
donald
trump
yes
it's
time
to
condemn
and
censure
adam
schiff
for
his
made-up
memo
on
the
conversation
between
president
trump
@lisamurkowski
what
will
you
do
go
to
the

the
western
world
realizes
being
used
as
an
excuse
to
start
a
war
with
iran
even
you
blamed
hillary
when
trump
did
it
last
time
repeating
repeatedly
saying
about
his
725
phone
conversation
wzelensky
trump
doesn't
need
us
the
if
more
offended
by
robert
de
niro
saying
the
word
fuck
than
you
are
with
the
fact
trump
is
fucking
our
country
then
our
kids
friends
and
neighbors
will
die
because
a
powerful
lobbying
group
hell
bent
on
unfettered
violence
and
a
bernie
kerik
is
a
convicted
felon
like
so
many
other
close
friends
of
donald
i
just
have
a
feeling
that
something
big
is
coming
and
i
want
everyone
to
be
very
very
careful
and
keep
each
other
if
we
are
gonna
start
looking
at
emails
gonna
need
the
trump
administration
to
hand
over
that
6
us
officials
other
than
president
were
implicated
in
the
whistleblower's
complaint
at
the
top
of
the
list
are
vp
pence
breaking
multiple
sources
have
confirmed
that
president
trump
plans
to
step
down
from
his
office
as
president
of
the
dude
is
broke
af
his


almost
look
forward
to
election
and
negate
the
votes
of
millions
of
evangelicals
in
the
process
they
know
the
only
impeachable
trump
says
he
wants
to
meet
whistleblower
warns
of
big
consequences
president
donald
trump
said
entitled
to
meet
the
ukraine
whistleblower
every
american
i
deserve
to
meet
my
this
rate
in
another
week
cognitive
decline
will
have
him
demanding
that
all
americans
strangle
and
we'll
re-elect
him
thanks
for
playing
cabinet
official
calls
us
racist
for
electing
trump
the
best
part
of
this
ukraine
phone
call
business
is
going
to
be
when
we
learn
us
presidents
have
been
burying
majority
of
americans
say
an
impeachment
inquiry
into
president
trump
is
necessary
a
new
poll
says
sometimes
a
thing
is
indefensible
and
inexplicable
henry
kissinger
is
a
top
advisor
to
putin
helped
normalize
trump
and
worked
with
russian
agent
dmitri
simes
in
trump
was
spied
on
and
accused
of
a
crime
he
immediately
released
proof
of
innocence
biden
was
accused
of
crimes
trump
did
nothing
wrong

are
deserting
him
fast
liberals
are
so
upset
with
trump
you
would
think
he
stole
90%
of
the
haitian
relief
fund
sold
20%
of
our
uranium
to
@sofyaangelz
straight
up
louisemensch
&amp;
her
teampatriot
work
for
murdoch
who
works
to
keep
@mollyjongfast
two
people
i
think
putin
pays
rohrabacher
and
kevin
mccarthy
said
in
june
2016
before
he
became
a
@speakerpelosi
if
the
it's
all
backfiring
politicians
hollywood
and
the
media
want
donald
trump
removed
from
office
because
he
is
an
existential
threat
to
their
trump
is
the
smoking
ukrainian
president
asked
about
the
aide
the
us
said
donald
trump
likes
to
boast
about
being
a
businessman
but
how
long
do
you
think
someone
acting
and
talking
like
him
would
@preetbharara
when
did
trump
learn
about
bidens
&amp;
ukraine
this
entire
impeachment
drama
around
trump
is
a
national
did
nothing
did
however
win
an
st
louis
post-dispatch
has
also
called
on
trump
to
resign
from
office
writing
time
has
come
for
republicans
to
stand
should
be
censured
by
the
hou

account
if
trump
goes
to
prison
it
will
be
his
first
complete
sentence
unknown\n'b'rt
\xe2\x80\x9cthis
is
about
proving
that
donald
trump
was
framed
by
the
@numbersmuncher
trump
is
not
fit
for
office
and
acts
more
impulsively
as
the
situation
worsens
for
long
past
time
for
at
least
the
30%
who
can't
read
apparently
life
after
trump
will
be
great
for
jamal
khashoggi
was
banned
from
his
own
country
for
criticizing
trump
and
was
epstein
had
the
goods
on
trump
hints
at
the
outbreak
of
a
war
like
if
removed
from
office
for
committing
serious
crimes
and
mahatma
great-grandson
attacks
father
of
nation
remark
you
know
what
i
love
about
trump
he
has
the
creative
authentic
balls
to
write
his
own
thoughts
and
share
them--
\xe2\x80\x9cfox
news
has
learned
that
the
pentagon
state
department
and
national
security
council
were
in
to
the
the
democrats
fool
trump
raised
up
to
13
million
dollars
since
pelosi
announced
the
impeachment
chris
wallace
is
playing
a
very
important
role
in
all
this
he
deserves

After breaking up the tweets into a list of individual words, we observe above that most of the noise has been eliminated using our string methods. As a consequence of using a list so large, there will still be a little bit of noise remaining that couldn't be accounted for. This is okay, and is to be expected. We'll consider that later in our analysis. But for now, just from eyeballing the list, it seems we cleaned up most of the document. 

In [29]:
len(trumplist)  #The total number of words in the tweet document

12740

Now we can begin using for loops and accumulators to iterate through the list of words in trumplist; and then count how many positive, negative, and stop words appear in Trump-related tweets. We'll create a fourth category called 'other words' to account for all words that are neither positive, negative, nor stop words. 

In [9]:
poswords=0
for word in trumplist:
    if word in poslist:
        poswords=poswords+1
        
negwords=0
for word in trumplist:
    if word in neglist:
        negwords=negwords+1
        
stopwords=0
for word in trumplist:
    if word in stoplist:
        stopwords=stopwords+1
        
otherwords=0
for word in trumplist:
    if word not in poslist and word not in neglist and word not in stoplist:
        otherwords=otherwords+1


In [10]:
poswords

797

In [11]:
negwords

594

In [12]:
stopwords

6485

In [13]:
otherwords

4998

Just from eyeballing our initial results, it appears that *Positive Words* $\gt$ *Negative Words*.
This is an interesting observation because it seems like 'Trump', being a politically charged keyword, would yield a more negative response rather than a positive one. 

Upon observing the list of positive keywords in positive.txt; we can see that the word *trump* (Not the individual) is listed as a positive word. This is a source of confusion for our analysis.
Let's count *positive words* one more time, now counting all positive words **Execept For** Trump.

In [14]:
poswords_updated=0
for word in trumplist:
    if word in poslist and word != 'trump':
        poswords_updated=poswords_updated+1

In [15]:
poswords_updated

340

Our number of positive words was cut in half. This yields a more reasonable result and ratio. We'll analyze our results at the end of our report.

##### Section 1.2: Using Our Own Key Word

In this section, I'll now extract a list of tweets using a keyword of my choice. I've chosen to use the keyword 'Canada'. The steps below are the exact same as the steps above and won't be elaborated upon. We'll analyze our results in the next section.

*Read the txt file*

In [16]:
canada = open('Canada.txt','r')
canada_content = canada.read()  
canada.close() 

*Preview the txt file*

In [17]:
canada_content

'b\'RT @StopTrudeau2019: @cathmckenna Can you please explain to Canadians why you are using these single use PLASTIC campaign signs? You and EV\\xe2\\x80\\xa6\\n\'b\'RT @HelenKennedy: Canada will pay for the moot.\\n\'b\'RT @Paola_Dec1231: Only virtue-signalling Toronto councilors and the clueless Trudeau Liberals will be surprised by this.\\n\\n87% of Ontarians\\xe2\\x80\\xa6\\n\'b\'RT @depressionnote: Suicide hotline numbers \\xf0\\x9f\\x93\\x9e \\n\\nUnited Kingdom \\xf0\\x9f\\x87\\xac\\xf0\\x9f\\x87\\xa7 116 123\\nUnited States \\xf0\\x9f\\x87\\xba\\xf0\\x9f\\x87\\xb8 1-800-273-8255\\nCanada \\xf0\\x9f\\x87\\xa8\\xf0\\x9f\\x87\\xa6 1 800 456 4566\\nIreland\\xe2\\x80\\xa6\\n\'b\'RT @cathmckenna: .@cathmckenna: Climate change is the urgent challenge of our time &amp; places a catastrophic debt on young people. It\\xe2\\x80\\x99s absol\\xe2\\x80\\xa6\\n\'b\'#HeyBangor, I have a road trip up to Canada and I\\xe2\\x80\\x99m swinging through town for lunch. What\\xe2\\x80\\x99s the hot s

*Make lower case and remove punctuation*

In [18]:
canada_content=canada_content.lower()

badchars='.,!?"///#[]()*$:'
for i in badchars:
    canada_content=canada_content.replace(i,'')
    

*Split into a list*

In [19]:
canadalist=canada_content.split()

*Remove further noise*

In [20]:
for i in canadalist:
    if i.startswith("https") or i.endswith("b'rt") or i.startswith("@"):
        canadalist.remove(i)
        
for i in canadalist:
    if "\\" in i or 'xe2' in i or 'xf' in i:
        canadalist.remove(i)
        
        
        
for i in canadalist:
    if i.find("\\") == True or i.find("xe2") == True or i.find("xf") == True:
        canadalist.remove(i)
        
        
        
for i in canadalist:
    if i.startswith("https"):
        canadalist.remove(i)
        
    if i.startswith("@"):
        canadalist.remove(i)
        

        

*Preview again*

In [21]:
for i in canadalist:
    print(i)

can
you
please
explain
to
canadians
why
you
are
using
these
single
use
plastic
campaign
signs
you
and
canada
will
pay
for
the
only
virtue-signalling
toronto
councilors
and
the
clueless
trudeau
liberals
will
be
surprised
by
of
suicide
hotline
numbers
\n\nunited
kingdom
116
states
1-800-273-8255\ncanada
1
800
456
climate
change
is
the
urgent
challenge
of
our
time
&amp;
places
a
catastrophic
debt
on
young
people
absol\xe2\x80\xa6\n'b'heybangor
i
have
a
road
trip
up
to
canada
and
swinging
through
town
for
lunch
the
hot
spot
dysarts
maximum
by
trudeau
&amp;
liberals
concluded
that
handguns
by
target
this
looks
like
the
fake
weather
from
disney
channel
we
purchased
a
backpack
in
texas
last
month
and
now
its
already
ripped
on
the
side
we
are
from
will
andrew
scheer
ruin
canada
via
of
british
columbia
karen
mckellin
international
leader
of
tomorrow
award
20202021
funded
study
this
is
what
real
leadership
looks
like
canada
needs
as
prime
minister
hey
deleted
you
deleted
your
poll
talk
about
why

to
ignore
this
photo
of
trudeau
wearing
an
afro
and
face
paint
while
working
at
the
didyouknow
only
35%
of
children
children
aged
5
to
17
get
enough
daily
physical
activity
an
easy
fix
walktoschool
intelligence
group
reports
that
one
of
the
masked
antifa
militants
who
harassed
the
elderly
woman
with
the
the
united
nations
wants
to
rule
you
and
have
both
signed
on
to
un
agenda
2030
no
information
on
asuran
canada
theatre
people
arriving
at
roxham
rd
are
told
that
are
coming
to
canada
illegally
as
soon
as
rcmp
touches
them
they
are
arrested
the
elderly
woman
with
a
walker
who
was
accosted
by
antifa
militants
in
hamilton
canada
has
spoken
out
in
a
new
clip
canada's
pierce
lepage
is
leading
the
men's
decathlon
long
jump
with
a
huge
779m
leap
comparing
canada
to
other
countries
without
oil
fair
at
all
no
matter
what
our
emissions
the
easiest
way
to
learn
about
political
regions
games
loved
seeing
my
students
so
engaged
today
with
bump
and
canada
gift
card
25
50
or
100
-
email
delivery
539
w

kenney
pleads
for
national
unity
during
speech
to
calgary
business
mr
premier
but
that
train
has
why
is
he
not
in
a
cell
waiting
see
clearly
now
that
police
services
have
no
recent
decision
in
pcl
constructors
westcoast
inc
v
royal
&amp;
sun
alliance
insurance
co
of
canada
was
would
so
love
to
go
but
canada
is
far
far
far
away
from
tulsa
ok
plus
the
ticket
price
is
out
of
my
price
does
the
uk
offer
of
post-brexit
temporary
tariff-free
access
to
87%
of
imports
still
noor
herbal
beauty
cream
anti-aging
free
phones
had
bootleg
access
to
google
apps
not
anymore
-
yahoo
canada
finance
via
nice
mention
about
our
upcoming
reading
high
plains
bookfest
to
feature
authors
from
canada
montana
in
annual
you
are
i
do
want
all
this
right
wing
fascist
reporting
to
end
this
election
will
let's
focus
on
our
triumph
in
the
faeroes
as
the
world
talks
about
gretathunberg
who
is
great
also
get
to
know
autumnpeltier
and
other
young
native
weaponizing
nickelback
the
government
of
canada
responds
a
city
counc

represented
in
parliament
should
be
included
in
debates
by
excluding
a
green
voice
tva
fails
@bonniecrombie
@oakvillemayor
stunning
hyperlapse
captures
huge
supercell
forming
in
alberta
canada
if
trudeau
wins
again
how
can
he
possibly
represent
canada
abroad
everyone
now
knows
a
joke
feeling
emotional
watching
the
absolute
sea
of
humanity
at
the
globalclimatestrike
march
in
toronto
so
many
young
kids
just
not
in
and
i'm
here
remembering
the
run
bts
ep
in
canada
where
jin
did
the
punishment
with
jimin
even
when
he
didn't
have
a
must
read
for
every
canadian
time
for
progressives
to
rally
around
the
to
insure
does
if
trudeau
wins
again
how
can
he
possibly
represent
canada
abroad
everyone
now
knows
a
joke
met
your
uncle
chris
last
summer
in
canada
a
ufo
sighting
has
been
commemorated
on
a
new
coin
issued
by
the
royal
canadian
mint
celebration
of
pop
culture
or
canada
needs
a
stronger
military
not
a
un
seat
maximum
by
trudeau
&amp;
liberals
concluded
that
handguns
by
target
weaponizing
nick

our
country
fuck
every
1001
within
an
hour
of
the
lennonwall
being
built
at
aberdeen
station
pro-ccp
mainland
chinese
nationalists
are
congratulates
wendy
brousseau
chia-yi
chua
nicolas
cloutier
geoff
hall
and
john
yuan
on
being
recognized
our
liberal
plant
2
billion
achieve
net-zero
emissions
by
ban
harmful
single-use
canada
you\xe2\x80\xa6
ready
rock
one
free
shipment
every
tuesday
in
october
even
come
pick
it
up
for
free
then
why
did
you
change
the
law
to
make
it
less
of
a
crime
to
be
involved
in
gang
activity
do
u
liberals
if
trudeau
wins
again
how
can
he
possibly
represent
canada
abroad
everyone
now
knows
a
joke
maximum
by
trudeau
&amp;
liberals
concluded
that
handguns
by
target
col\xe2\x80\xa6\n'brt
we're
proud
of
our
amazing
slate
of
candidates
who
are
truly
representative
of
canada
and
who
will
stand
up
for
people
across
keep
campaigning
the
photo
op
with
on
the
back
of
a
truck
showed
real
sensitivity
to
climate
this
guy
pm
closest
competitor
in
election
he
is
a
mini-trump
a
se

cbc
news
division
-
responsibly
canada
@jkenney
canada\xe2\x80\x99s
economy
flat
lined
in
july
largely
because
of
a
decline
in
the
energy
to
believe
that
there
are
can
they
go
to
canada
already
ffssssss
hopefully
when
im
there
in
april
or
in
the
summer
this
election
is
not
about
right
vs
left
it
is
about
globalism
vs
national
sovereignty
we
are
the
only
main
party


In [28]:
len(canadalist) #total word amount on tweet document

17052

*Count each category*

In [23]:
poswords_canada=0
for word in canadalist:
    for word1 in poslist:
        if word1==word:
            poswords_canada=poswords_canada+1
  
negwords_canada=0
for word in canadalist:
    if word in neglist:
        negwords_canada=negwords_canada+1
        
stopwords_canada=0
for word in canadalist:
    if word in stoplist:
        stopwords_canada=stopwords_canada+1
        
otherwords_canada=0
for word in canadalist:
    if word not in poslist and word not in neglist and word not in stoplist:
        otherwords_canada=otherwords_canada+1

*Observe amount of each word category*

In [24]:
poswords_canada

666

In [25]:
negwords_canada

496

In [26]:
stopwords_canada

8405

In [27]:
otherwords_canada

7694

#### Section 2: Sentiment Analysis 

##### Section 2.1: Analyzing Trump Tweets

**Note: The total word amount may differ slightly (by about 0.5 - 1%) each time the program is run. Perhaps due to the size of the txt files. I tried setting a seed to form a consistent number generation. Let's just take into consideration this approximately 1% difference before beginning our analysis. The information below is based on when I initially ran the program, and numbers may differ slightly at later run-times.**

For part 1, we found 797 positive words, 594 negative words, 6485 stops words, and 4829 'other words' (words that don't fall into the first three categories). This gives a total sum of 12,705 words in our entire database of Trump-related tweets. Using the length function, we yielded a total of 12,571 words in our Trump tweet database. These two values differ. But we'll neglect that for now and just say it is a consequence of a bit of disparity between the string methods we used to clean up our document, and the iteration and counters in our for loops. This slight disparity is to be expected from such a large list of words. This comes out to only about a **1**$\%$ difference. So we'll neglect this for now.

Initially, I found it odd that there were more positive words that negative words, espescially considering the politically-charged nature of our keyword. I then realized that it was an overestimate since the word 'trump' is included in our list of positive words. Neglecting that word, our total amount of positive words was cut in half to 340 words, which makes more sense.

The ratio of negative to positive words (594/340) is approximately 1.7;
This is close to 2, which means that this keyword yields a rather significantly negative sentiment. 
When finding the ratio of negative to total words (594/12571), this tells us that approximately 5$\%$ of all words are negative.
Stops words (6475/12571) account for about 52$\%$ of the document, accounting for the majority. 
Other words (4829/12571) comporise 38$\%$. Both categories together make up a **large** majority of the document. Meaning that even though negative words make up a much smaller piece of the pie, **we are weighing them much more heavily since this is a sentiment analysis.**


The sum of positive and negative words is 934, accounting for only about one percent of the document. But again, they are being weighed more heavily since this is a sentiment analysis. 

The difference between positive and negative (340 - 594) yields -254. This gives us a net negative sentiment. This, along with the fact that our ratio of negative to positive equates to 1.7; tells us that this is a rather significantly negative sentiment.

##### Section 2.2: Analyzing Canada-related Tweets

For part 2, we found 666 positive words, 496 negative words, 8405 stop words and 7546 other words. This gives a total sum of 17,113 words. The length function gave us 16,904 words. This disparity, again, only comes out to around 1%.

This document gave us more positive words than negative words. The ratio of positive to negative words (666/496) is 1.3; which is close to 1, which means that this keyword yields a slightly positive sentiment. The ratio of positive to total words (666/16904) tells us that about 4% of all words are positive. Stop words (8405/16904), again, account for about 50% of the document. Other words (7546/16904) make up 44%. Both categories together, again give us a percentage in the nineties. 

The sum of positive and negative words is 1162, again account for less than one percent of the document, but weighed much more heavily than the filler words. 

The difference between positive and negative (666 - 496) is 170. This is a net positive sentiment. This, along with the fact that our ratio of positive to negative equate to 1.3; tells us that this a slightly positive sentiment. I believe this yielded an overall more positive sentiment than our first keyword because the keyword 'Canada' is much less politically-charged. The word 'Trump' is very attached to politics and current events which are much more hot button issues. 